pkgs/development/libraries/ncurses/default.nix

```diff
@@ -12,6 +14,12 @@
 , buildPackages
 }:
 
+let
+  soname_tinfo = if onlytinfo
+    then "tinfo"
+    else "tinfow";
```

Comment:
I'm kinda blindly copying this idiom from the OpenSUSE packaging docs, but it seems like the only "sane" way to handle this situation where something else might also be providing `libtinfo` (unless I'm misunderstanding something here).

Reply:
Surely you meant to check for `unicode`? That’s the difference between `libncurses` and `libncursesw`.

But, `libtinfo` and `libtinfow` are supposed to be compatible, so that a separate `libtinfow` is not necessary. There have been bugs with this in the past, but they were treated as bugs by the upstream maintainer and fixed (https://bugzilla.redhat.com/show_bug.cgi?id=1580323).

Reply:
Thanks for this; I'm pretty unfamiliar with ncurses packaging and I've been piecing things together today in an attempt to workaround some local tooling packaging.

I've dropped this, so at the moment it seems as if there's nothing useful in this PR at all; I'm inclined to close it since I can't figure out whether `--with-versioned-symbols` actually resolves my issue or not.

Is there anything else you would need to help get #95141 finalized?